text_generation/causal_lm/cpp/beam_search_causal_lm.cpp

```diff
@@ -89,5 +89,5 @@ int main(int argc, char* argv[]) try {
     return 1;
 } catch (...) {
     std::cerr << "Non-exception object thrown\n";
-    return 1;
+    return 2;
```

Comment:
should we be consistent and return either numbers or `EXIT_FAILURE` / `EXIT_SUCCESS` ?